### DISCUSSION TITLE
Changing tracks names

### DISCUSSION THREAD

**Discussion by coeur-de-loup** [leolabs/ableton-js#62] [Q&A]
Hi, 

First of all, this project is awesome, thanks so much for your hard work and for sharing it!

I managed to make the test() script work, but I can't find the right syntax to change a track name... Is it possible?

**Comment by coeur-de-loup** [comment]
I found the problem. I was trying to access the track by its name instead of its index! Is there a way to get track by name, or we have to loop over all tracks to find index?

`const tracks = await ableton.song.get("tracks");`
  `tracks[0].set("name", "hello world")`

**Comment by leolabs** [comment]
Hey @coeur-de-loup,

Looping through the tracks is currently the way to go since Live returns an array of tracks. I'd recommend using the `find` method for this, like:

```ts
const tracks = await ableton.song.get("tracks");
const track = tracks.find(t => t.raw.name === "My Track");
track?.set("name", "My Track 2");
```

I hope this helps!

**Reply by coeur-de-loup** [reply]
Hey @leolabs! I just saw your answer, thanks so much for the help!
